test/controllers/IllnessStartDateController.test.ts

```diff
@@ -0,0 +1,156 @@
+import { expect } from 'chai';
+import { MOVED_TEMPORARILY, OK, UNPROCESSABLE_ENTITY } from 'http-status-codes';
+import { before } from 'mocha';
+import request from 'supertest';
+
+import 'app/controllers/EvidenceDownloadController';
+import {
+    ENTRY_PAGE_URI,
+    ILLNESS_START_DATE_PAGE_URI
+} from 'app/utils/Paths';
+
+import { createApp } from 'test/ApplicationFactory';
+
+const pageHeading: string = 'When did the illness start?';
+const errorSummaryHeading: string = 'There is a problem';
+const invalidStartDayErrorMessage: string = 'You must enter a day';
+const invalidStartMonthErrorMessage: string = 'You must enter a month';
+const invalidStartYearErrorMessage: string = 'You must enter a year';
+const invalidDateErrorMessage: string = 'Enter a real date';
+const invalidDateFutureErrorMessage: string = 'Start date must be today or in the past';
+
+describe('IllnessStartDateController', () => {
+
+    describe('GET request', () => {
+
+        before(() => {
+            process.env.ILLNESS_REASON_FEATURE = '1';
+        });
+
+        it('should return 200 when trying to access the page', async () => {
+            const app = createApp();
+            await request(app).get(ILLNESS_START_DATE_PAGE_URI).expect(response => {
+                expect(response.status).to.be.equal(OK);
+                expect(response.text).to.contain(pageHeading);
+            });
+        });
+
+        it('should redirect to entry page when illness reason feature is disabled', async () => {
+
+            process.env.ILLNESS_REASON_FEATURE = '0';
+
+            const app = createApp();
+            await request(app).get(ILLNESS_START_DATE_PAGE_URI)
+                .expect(response => {
+                    expect(response.status).to.be.equal(MOVED_TEMPORARILY);
+                    expect(response.header.location).to.include(ENTRY_PAGE_URI);
+                });
+        });
+    });
+    describe('POST request', () => {
+
+        before(() => {
+            process.env.ILLNESS_REASON_FEATURE = '1';
+        });
```

Comment:
does this need to be set before every test? or could it just be:

```suggestion
        process.env.ILLNESS_REASON_FEATURE = '1';
```

Reply:
i've added a before() and after() to retain the feature state at class level, and have then added a beforeEach() to enable to feature for the tests